### ISSUE TITLE
Formatter: String interpolation with macro expansion

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#9154]
When a string interpolation contains a macro expansion, it results in three consecutive curly braces: `"#{{{ macro_expression }}}"`.  Having spaces in between helps reading the code: `"#{ {{ macro_expression }} }"`.
The formatter currently removes those spaces. I think it should rather add them when missing to improve readability.

**Comment by waj** [comment]
Should it add space also for other expressions with curly braces? (tuples, hashes)

**Comment by Blacksmoke16** [comment]
Should we also go as far as to replace `{{macro_expression}}` with `{{ macro_expression }}`?

**Comment by Blacksmoke16** [comment]
Is this something we're wanting to put behind a feature flag for now? The diff is pretty big, just a lot of files: https://github.com/crystal-lang/crystal/compare/master...Blacksmoke16:spaces-around-macro-controls.

**Comment by straight-shoota** [comment]
Yes, adding spaces inside the macro expression should be opt-in. It has significant impact and might spark some discussion.
I think this is independent of this issue here, though. So it should be a different change.
Space between curly braces is important for disambiguation, but space inside the curly braces is just stylistic.
